lab_3_keywords_textrank/main.py

```diff
@@ -302,14 +356,17 @@ def fill_positions(self, tokens: tuple[int, ...]) -> None:
             tokens : tuple[int, ...]
                 sequence of tokens
         """
-        pass
+        for count, item in enumerate(tokens):
+            self._positions[item] = self._positions.get(item, []) + [count + 1]
 
     # Step 8.3
     def calculate_position_weights(self) -> None:
         """
         Computes position weights for all tokens in text
         """
-        pass
+        non_normalized = {token: sum(1 / position for position in self._positions[token]) for token in self._positions}
```

Comment:
use `.items()` in the external loop to get keys and values -> so you do not need `self._positions[token]` in the internal loop